handlers/TokenHandler.go

```diff
@@ -37,6 +37,42 @@ type PublicTokenResponse struct {
 	PriceData map[string]string `json:"price_data"`
 }
 
+func (token *TokenHandler) Index(c echo.Context) error {
+	userData, err := services.AuthUser(c, *token.server.DB)
+	if err != nil {
+		return c.JSON(http.StatusNotFound, map[string]string{
+			"error": err.Error(),
+		})
+	}
+	var userTokens []models.Token
+	token.server.DB.Where("user_id = ?", userData.ID).Table("tokens").Find(&userTokens)
+
+	return c.JSON(http.StatusOK, UserTokenIndexResponse{userTokens})
+}
+
+type UserTokenIndexResponse struct {
+	Token []models.Token `json:"tokens"`
+}
+
+func (token *TokenHandler) AirdropClaimAmount(c echo.Context) error {
+	tokenId, err := strconv.Atoi(c.Param("id"))
+	airdropData, err := models.GetAirdropByTokenId(uint64(tokenId), *token.server.DB)
+	if err != nil {
+		return c.JSON(http.StatusNotFound, map[string]string{
+			"error": "no airdrop available for the given tokenId",
+		})
+	}
+	wallet_address := c.Param("address")
+	airdropClaimData, err := models.GetClaimAmountByAirdropID(uint64(airdropData.ID), wallet_address, *token.server.DB)
+	if err != nil {
+		return c.JSON(http.StatusNotFound, map[string]string{
+			"error": "there is no airdrop available for your address",
```

Comment:
This error is a bit misleading. Might be more accurate to say something about 'this' airdrop.
"This airdrop is unavailable for this address"